### PR TITLE
Feature/data export api

### PR DESCRIPTION
**Pull request by jaysencpp**

### CODE REVIEW

File: src/http/hooks/dataExport/useExcelExport.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                const {
+                    data: { tempId },
+                } = await dataExport.createExcelFile({
+                    fileName,
+                    dataSetName,
+                    sheets: data.sheets,
+                });
+                setUrl(`${resourceCollections.dataExport.download(tempId)}.xlsx`);
+            } catch (err) {
+                console.error(err);
```
Comment (jaysencpp) [inline src/http/hooks/dataExport/useExcelExport.ts:48]: Handle this? Just return it from the hook?

File: src/http/apiClients/FusionClient.ts [hunk 2]
```diff
@@ -54,4 +56,59 @@ export default class FusionClient extends BaseApiClient {
...
+
+    public createExcelFile(excelData: DataExportRequest) {
+        const url = this.resourceCollections.fusion.exportExcel();
+        return this.httpClient.postAsync<
+            DataExportRequest,
+            DataExportResponse,
+            FusionApiHttpErrorResponse
+        >(url, excelData);
+    }
+
```
Comment (odinr) [inline src/http/apiClients/FusionClient.ts:68]: add tsdoc

File: src/http/apiClients/FusionClient.ts [hunk 3]
```diff
@@ -54,4 +56,59 @@ export default class FusionClient extends BaseApiClient {
             FusionApiHttpErrorResponse
         >(url, undefined);
     }
+
```
Comment (odinr) [inline src/http/apiClients/FusionClient.ts:59]: add tsdoc

File: src/http/hooks/dataExport/useExcelExport.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ *  )
+ * }
+ *
+ */
+export const useTableExport = (props: { fileName: string; dataSetName: string }) => {
+    const { fusion } = useApiClients();
+
+    const { fileName, dataSetName } = props;
+    const client = useCallback(
+        async (data: { sheets: Sheet[] }) => {
```
Comment (odinr) [inline src/http/hooks/dataExport/useExcelExport.ts:32]: remove async and type hint return type

File: src/http/hooks/dataExport/useExcelExport.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            return await fusion.getExcelStatusInterval({
```
Comment (odinr) [inline src/http/hooks/dataExport/useExcelExport.ts:33]: no point awaiting, return the method

**Review by odinr — CHANGES_REQUESTED** [review]

**Pull request merged**